inst/include/RcppSimdJson/deserialize/matrix.hpp

```diff
@@ -64,9 +64,9 @@ inline Rcpp::Vector<RTYPE> build_matrix_typed(simdjson::dom::array array,
 
 #else
 
-    for (auto&& sub_array : array) {
+    for (simdjson::dom::array sub_array : array) {
```

Comment:
This new code throws if array is not made of arrays. It used to do something unsafe in such cases.

Note that this whole function is noexcept(RCPPSIMDJSON_NO_EXCEPTIONS) and I have definitively introduced some exception throwing code in there. I think it means that you get a violent abort in case of error.

Reply:
I think that is what the SAN run triggered too.  We may be able to relax some of the `noexcept`.  

Looking good overall though!  Thanks for all your work here.  Brendan is still 'under water' for a little bit but will surface it due course.

Reply:
I can easily do exception-less code by doing...

```C++
for(auto element : array) {
if(simdjson::dom::array sub_array; element.get(subarray) == simdjson::SUCCESS) {
   for (simdjson::dom::array sub_array : array) {
     ....
   }
} else {
// something is wrong
}

}
```

Safe, exceptionless code is unavoidably more verbose because I need to catch the error. The code that this is replacing did not handle error condition, so I felt that throwing was the right equivalent. 

Throwing inside a noexcept should trigger an abort which is sometimes the right move (better than silently continuing).

In this case, I think that an exception/error would signal a bug in the library... meaning that there is no recovery possible other than fixing the code.